cpp/src/arrow/compute/kernels/hash_aggregate.cc

```diff
@@ -1866,17 +1940,40 @@ Result<std::unique_ptr<KernelState>> MinMaxInit(KernelContext* ctx,
 
 struct GroupedMinMaxFactory {
   template <typename T>
-  enable_if_number<T, Status> Visit(const T&) {
+  enable_if_physical_integer<T, Status> Visit(const T&) {
```

Comment:
Same remark as for the scalar aggregations: this will generate a separate kernel per logical type?